modules/minimap2/1.0/minimap2.smk

```diff
@@ -0,0 +1,168 @@
+
+# To avoid this we need to add the "packaging" module as a dependency for LCR-modules or oncopipe
+
+current_version = pkg_resources.get_distribution("oncopipe").version
+if version.parse(current_version) < version.parse(min_oncopipe_version):
+    print('\x1b[0;31;40m' + f'ERROR: oncopipe version installed: {current_version}' + '\x1b[0m')
+    print('\x1b[0;31;40m' + f"ERROR: This module requires oncopipe version >= {min_oncopipe_version}. Please update oncopipe in your environment" + '\x1b[0m')
+    sys.exit("Instructions for updating to the current version of oncopipe are available at https://lcr-modules.readthedocs.io/en/latest/ (use option 2)")
+
+# End of dependency checking section
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["minimap2"]`
+CFG = op.setup_module(
+    name = "minimap2",
+    version = "1.0",
+    # TODO: If applicable, add more granular output subdirectories
+    subdirectories = ["inputs", "minimap2", "sort_bam", "outputs"],
+)
+
+#include: "../../utils/2.1/utils.smk"
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _minimap2_input_fastq,
+    _minimap2_symlink_bam,
+    _minimap2_output_bam,
+    _minimap2_all,
+
+sample_ids_minimap2 = list(CFG['samples']['sample_id'])
+
+##### RULES #####
+
+
+# Symlinks the input files into the module results directory (under '00-inputs/')
+rule _minimap2_input_fastq:
+    input:
+        fastq = CFG["inputs"]["sample_fastq"]
+    output:
+        fastq = CFG["dirs"]["inputs"] + "fastq/{seq_type}/{sample_id}.fastq.gz"
+    run:
+        op.absolute_symlink(input.fastq, output.fastq)
+
+
+rule _minimap2_run:
+    input:
+        fastq = str(rules._minimap2_input_fastq.output.fastq),
+        fasta = reference_files("genomes/{genome_build}/genome_fasta/genome.fa")
+    output:
+        sam = pipe(CFG["dirs"]["minimap2"] + "{seq_type}--{genome_build}/{sample_id}_out.sam")
+    log:
+        stdout = CFG["logs"]["minimap2"] + "{seq_type}--{genome_build}/{sample_id}/minimap2.stdout.log",
+        stderr = CFG["logs"]["minimap2"] + "{seq_type}--{genome_build}/{sample_id}/minimap2.stderr.log"
+    params:
+        opts = CFG["options"]["minimap2"]
+    conda:
+        CFG["conda_envs"]["minimap2"]
+    threads:
+        CFG["threads"]["minimap2"]
+    resources:
+        **CFG["resources"]["minimap2"]
+    shell:
+        op.as_one_line("""
+        minimap2 {params.opts}
+        -t {threads} 
+        {input.fasta} 
+        {input.fastq} 
+        -o {output.sam} 
+        > {log.stdout} 
+        2> {log.stderr}
+        """)
+
+
+rule _minimap2_samtools:
+    input:
+        sam = str(rules._minimap2_run.output.sam)
+    output:
+        bam = CFG["dirs"]["minimap2"] + "{seq_type}--{genome_build}/{sample_id}_out.bam",
+        complete = touch(CFG["dirs"]["minimap2"] + "{seq_type}--{genome_build}/{sample_id}_out.bam.complete")
```

Comment:
Not sure how this works here because this output is not used in the shell command or elsewhere in the module. Maybe this is left from the earlier versions and needs to be dropped?

Reply:
Snakemake uses the touch flag to touch an empty file without having to write a shell command for it https://snakemake.readthedocs.io/en/master/snakefiles/rules.html#flag-files

Reply:
Kostia's point is that writing the `.complete` file alone isn't enough. For it to have the desired effect (forcing this rule to re-run if it's interrupted/the bam file is incomplete) it has to be an input to a subsequent rule. Can you add the `complete` file from this rule as an input to the next rule in the module?